cmd/provider/main.go

```diff
@@ -51,7 +51,7 @@ func main() {
 		maxReconcileRate        = app.Flag("max-reconcile-rate", "The maximum number of concurrent reconciliation operations.").Default("1").Int()
 		artifactsHistoryLimit   = app.Flag("artifacts-history-limit", "Each attempt to run the playbook/role generates a set of artifacts on disk. This settings limits how many of these to keep.").Default("10").Int()
 		pollStateMetricInterval = app.Flag("poll-state-metric", "State metric recording interval").Default("5s").Duration()
-		replicasCount           = app.Flag("replicas", "Amount of replicas configured for the provider. When using more than 1 replica, reconciles will be sharded across them based on a modular hash.").Default("1").Int()
+		replicasCount           = app.Flag("replicas", "Amount of replicas configured for the provider. When using more than 1 replica, reconciles will be sharded across them based on a modular hash.").Default("1").Uint32()
```

Comment:
golangci-lint was complaining about `int`-`uint32` conversion later in the code, which has the potential to overflow. I 
changed all intermediate and this one "source" var relevant to this case to be `uint32` in the first place. They are all things like replaca counts, number of shards and shard indices and cannot be negative.